eqeq bug

MSVC gives a warning about:
src\charges\eqeq.cpp(98): warning C4805: '!=' : unsafe mix of type 'bool' and type 'const double' in operation

    if (!_ionizations[1][0] != hi_0)

Double negative? (Pun intended)

--- a/src/charges/eqeq.cpp
+++ b/src/charges/eqeq.cpp
@@ -95,9 +95,16 @@ namespace OpenBabel
     OBAtom *atom;
 
     // If parameters have not yet been loaded, do that
-    if (!_ionizations[1][0] != hi_0)
-      if (!ParseParamFile())
+    if (!_paramFileLoaded)
+    {
+      if (ParseParamFile())
+      {
+        _paramFileLoaded = true;
+      } else
+      {
         return false;
+      }
+    }
 
     // Calculate atomic properties based around their ionic charge
     for (i = 0; i < N; i++)

--- a/src/charges/eqeq.h
+++ b/src/charges/eqeq.h
@@ -30,7 +30,9 @@ GNU General Public License for more details.
 #include <Eigen/SVD>
 
 #define TABLE_OF_ELEMENTS_SIZE 84 // Number of atoms in data/eqeqIonizations.txt
-#define PI 3.1415926535897932384626433832795 // 32 digist of pi
+#define PI 3.1415926535897932384626433832795 // 32 digits of pi
+
+static bool _paramFileLoaded = false; // Flipped after parameter file is loaded
 
 // The following constants are functionalized in EQeq's original implementation,
 // but rarely ever changed in practice.

